src/KDUtils/signal.h

```diff
@@ -0,0 +1,26 @@
+#ifndef KDUTILS_SIGNAL_H
+#define KDUTILS_SIGNAL_H
+
+#include <kdbindings/signal.h>
+
+using namespace KDBindings;
+
+namespace KDUtils {
+
+template<typename... Args>
+class Signal : public KDBindings::Signal<Args...>
```

Comment:
Not sure we need this subclass. Can't the user just call connectDeferred() directly?

I'm thinking of the case where you add loads of KDBindings::Signals to your classes, then realise you need to replace them with KDUtils::Signal. To me it makes it clearer what the intent is as well as less refactoring to just use KDBindings::Signal and call connectDeferred().

But maybe I missed something subtle?

Reply:
Thank You Sean for looking at the patch! Yeah, I'm also not exactly sure about this, similarly we can have several benefits, if we need functionality specific to KDUtils use cases then we have the liberty to modify the KDUtils::Signal without modifying KDBindings::Signal, it may be additional logging/debugging, or additional thread safety measures. But the other approach is fine to me as well.